### PR TITLE
swagger api documentation

### PR DESCRIPTION
**Pull request by gracemugwanezak**
## Purpose
made to integrate swagger api documentation in the mavericks e-commerce

## Changes Made
added a `docs` folder to wrap up a .`yaml` file for the documentation

## Testing Instructions

no test needed

## Related Issues

no issues yet

## Checklist

Please review the following checklist and make sure all tasks are complete before submitting:

- [x] Code follows the project's coding standards
- [x] Changes are covered by tests
- [x] Documentation is updated (if applicable)
- [x] All tests pass

### CODE REVIEW

**Review by Favor-star — APPROVED** [review]
Looks goods to me!

File: src/server.ts [hunk 1]
```diff
@@ -13,3 +18,27 @@ app.get("/", async (req: Request, res: Response): Promise<Response> => {
 app.listen(PORT, () => {
   console.log(`Server is running on port ${PORT}`);
 });
+
+const options = {
```
Comment (patrickhag) [inline src/server.ts]:
```suggestion
const options = {
```

I guess this options should be in it's separate file,  that will look way more cleaner

**Review by patrickhag — COMMENTED** [review]
maybe you can consider those thing

**Review by amin-leon — APPROVED** [review]

**Review by P-Rwirangira — APPROVED** [review]

**Comment by niyontwali** [comment]
Please resolve the conflicts

**Review by patrickhag — APPROVED** [review]

**Review by niyontwali — APPROVED** [review]
Well done

**Pull request merged**